OneCommunityGlobal/HGNRest
# Johny fix bug badges I have earned

# Description
Fixes Bug from the badge bug list 4 - (PRIORITY MEDIUM) : Below are the badges I’ve earned. This should represent 1 time I got three 30-hr weeks in a row and 2 times I got two 30-hr weeks in a row.
![Captura de Tela (330)](https://user-images.githubusercontent.com/39320057/232641759-41b1f682-eef2-4740-9ecb-8ef7b27975ca.png)



## Mainly changes explained:
1 - The function `checkxHrsforXWeeks` was rebuild.
2 -  Now thw badges are being assigned properly.
3 - Badges of lower streak now are being delete when a badge of greate streak is earned.

## How to test:
check into current branch
do `npm install` and `npm run build` to run this PR locally
:arrow_right: go to the `useProfileJobs.js ` file and it is necessary to set the cronjob to run  in a short interval of time
:white_check_mark: for example to run the cronjob every 30 seconds use :  `'*/30 * * * * *'`.
:arrow_right:  to assign badges for 1 week or weeks greater than 1 it is necessary the user had worked for a entire week,  because the badges is assigned based on last week tangible hours.
:arrow_right: assign tangible hours for the user in any project.

![Captura de Tela (331)](https://user-images.githubusercontent.com/39320057/232643907-e8460812-0dd3-4841-94cb-56f2c71445c0.png)

:warning: :warning:
## it is not a easy PR to test
